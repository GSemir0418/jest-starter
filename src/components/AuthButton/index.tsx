import React, { FC, useEffect, useState } from "react";
import { Button, ButtonProps, message } from "antd";
import classnames from "classnames";
import styles from "./AuthButton.module.less";
import { getUserRole, UserRoleType } from "apis/user";

type Props = ButtonProps;

const mapper: Record<UserRoleType, string> = {
  user: "普通用户",
  admin: "管理员",
};

const AuthButton: FC<Props> = (props) => {
  const { children, className, ...restProps } = props;

  const [userType, setUserType] = useState<UserRoleType>();

  // 获取用户身份并设置
  const getLoginState = async () => {
    const res = await getUserRole();
    setUserType(res.data.userType);
  };

  useEffect(() => {
    getLoginState().catch((e) => message.error(e.message));
  }, []);

  return (
    <Button {...restProps} className={classnames(className, styles.authButton)}>
      {mapper[userType!] || ""}
      {children}
    </Button>
  );
};

export default AuthButton;
