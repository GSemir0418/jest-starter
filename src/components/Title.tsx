import React from 'react'
export const Title = (props: { title: string }) => {
  const { title } = props
  return (
    <p style={{ fontSize: '20px', color: 'red' }}>{title}</p>
  )
}