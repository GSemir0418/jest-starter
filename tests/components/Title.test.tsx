import { render } from "@testing-library/react";
import { Title } from "components/Title";
import React from 'react'

describe("Title", () => {
  it('可以正确渲染标题', () => {
    const { baseElement } = render(<Title title="大字" />);
    expect(baseElement).toMatchSnapshot();
  })
})