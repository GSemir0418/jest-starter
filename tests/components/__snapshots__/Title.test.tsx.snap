// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title 可以正确渲染标题 1`] = `
<body>
  <div>
    <p
      style="font-size: 20px; color: red;"
    >
      大字
    </p>
  </div>
</body>
`;
